tck/src/main/java/org/eclipse/microprofile/lra/tck/TckTests.java

```diff
@@ -402,7 +402,7 @@ private void joinAndEnd(boolean close, String path, String path2)
         int completionCount = lraMetricService.getMetric(LRAMetricType.Completed, lra, LraResource.class.getName());
         int compensationCount = lraMetricService.getMetric(LRAMetricType.Compensated, lra, LraResource.class.getName());
 
-        boolean wasCalled = (close ? completionCount > 0 : compensationCount > 0);
+        boolean wasCalled = (close ? completionCount == 1 : compensationCount == 1);
```

Comment:
You cannot guarantee that messages will not be replayed. In a distributed system "at least once" semantics is normal (the way the test was originally coded). "Exactly once" (the way you have just coded the test) is not feasible. And "at most once" is not a useful delivery guarantee for our spec so we don't need to discuss it.

Reply:
this method is used only in acceptCloseTest and acceptCancelTest which both invoke LraResource which has a Status method. The spec says "But if the method `compensatePreviousAction` should not be called a
second time (i.e. it is not idempotent) then the participant has the option of
reporting its` progress using the `@Status` annotation" (since there is no better citation to be used, we should add text that explicitly states it).

The Status method is in my understanding (and above citation) a guarantee that the implementation will not invoke Complete or Compensate methods several times since they don't need to be idempotent. Are you saying that LRA does not provide this guarantee?

Reply:
I am saying that it is not obvious why the assert is asserting only 1 call without me having to inspect the resources and consult the specification. The assert message should describe why it only expects 1 callback.